### PR TITLE
Revert combination mode to before import_region call

### PR DESCRIPTION
**Pull request by javerbukh**
### Description

This pull request reverts the combination mode to before the `import_region(region, combination_mode=<list>)` call.

### Change log entry

- [ ] Is a change log needed? If yes, is it added to `CHANGES.rst`? If you want to avoid merge conflicts,
  list the proposed change log here for review and add to `CHANGES.rst` before merge. If no, maintainer
  should add a `no-changelog-entry-needed` label.

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [ ] Are two approvals required? Branch protection rule does not check for the second approval. If a second approval is not necessary, please apply the `trivial` label.
- [ ] Do the proposed changes actually accomplish desired goals? Also manually run the affected example notebooks, if necessary.
- [ ] Do the proposed changes follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are tests added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Are docs added/updated as required? If so, do they follow the [STScI Style Guides](https://github.com/spacetelescope/style-guides)?
- [ ] Did the CI pass? If not, are the failures related?
- [ ] Is a milestone set? Set this to bugfix milestone if this is a bug fix and needs to be released ASAP; otherwise, set this to the next major release milestone. Bugfix milestone also needs an accompanying backport label.
- [ ] After merge, any internal documentations need updating (e.g., JIRA, Innerspace)?

### CODE REVIEW

**Comment by pllim** [comment]
Is this testable as a unit test?

File: jdaviz/configs/default/plugins/subset_plugin/subset_plugin.py [hunk 1]
```diff
@@ -967,6 +970,8 @@ def _load_regions(self, regions, combination_mode=None, max_num_regions=None,
             n_loaded += 1
             if max_num_regions is not None and n_loaded >= max_num_regions:
                 break
+        # Revert edit mode to before the import_region call
+        self.app.session.edit_subset_mode.mode = previous_mode
```
Comment (kecnry) [inline jdaviz/configs/default/plugins/subset_plugin/subset_plugin.py:974]: is there anything above that could actually raise an error (not caught by `bad_regions`) where the revert should still happen?

Comment (javerbukh) [reply jdaviz/configs/default/plugins/subset_plugin/subset_plugin.py:974]: Not that I could see, does anything stick out to you?

File: jdaviz/configs/default/plugins/subset_plugin/tests/test_subset_plugin.py [hunk 2]
```diff
@@ -198,6 +199,7 @@ def test_import_spectral_region(cubeviz_helper, spectrum1d_cube, spec_regions, m
     subsets = cubeviz_helper.app.get_subsets()
     assert len(subsets) == len_subsets
     assert len(subsets['Subset 1']) == len_subregions
+    assert cubeviz_helper.app.session.edit_subset_mode.mode == ReplaceMode
```
Comment (javerbukh) [inline jdaviz/configs/default/plugins/subset_plugin/tests/test_subset_plugin.py:202]: @pllim I added this and the lines below, let me know if that is sufficient coverage.

File: jdaviz/configs/default/plugins/subset_plugin/subset_plugin.py (hunk 1 above)
Comment (kecnry) [reply jdaviz/configs/default/plugins/subset_plugin/subset_plugin.py:974]: I guess we don't need to catch bugs in the code itself, so `bad_regions` should be sufficient to make sure the revert happens here.  Thanks!

**Review by kecnry — APPROVED** [review]
LGTM

**Review by pllim — APPROVED** [review]
Thanks!

**Pull request merged**